buildtools/npm/manifest.go

```diff
@@ -35,3 +65,101 @@ type Lockfile struct {
 func FromLockfile(filename string) (Lockfile, error) {
 	return Lockfile{}, errors.ErrNotImplemented
 }
+
+func convertManifestToPkg(manifest manifest) pkg.Package {
+	id := pkg.ID{
+		Type:     pkg.NodeJS,
+		Name:     manifest.Name,
+		Revision: manifest.Version,
+	}
+
+	var imports pkg.Imports
+	for depName, version := range manifest.Dependencies {
+		imports = append(imports, createImport(depName, version))
+	}
+
+	return pkg.Package{
+		ID:      id,
+		Imports: imports,
+	}
+}
+
+func createImport(packageName string, version string) pkg.Import {
+	id := pkg.ID{
+		Type:     pkg.NodeJS,
+		Name:     packageName,
+		Revision: version,
+	}
+
+	return pkg.Import{
+		Target:   packageName,
+		Resolved: id,
+	}
+}
+
+func fromSubNodeModules(currentDir string, rootNodeModuleDir string, previousPackage pkg.Package) (map[pkg.ID]pkg.Package, error) {
```

Comment:
This logic is difficult to follow, and I think it's because you're doing raw recursion. I would recommend using `filepath.Walk` instead.